test/modules/geo-layers/tile-layer/tile-2d-header.spec.js

```diff
@@ -65,3 +67,104 @@ test('Tile2DHeader#Abort quickly', async t => {
   t.notOk(requestScheduler.requestMap.has(tile), 'Scheduler deletes tile on abort');
   t.end();
 });
+
+const LOCAL_TILE_COORDS = [
+  {
+    type: 'Feature',
+    geometry: {
+      type: 'Polygon',
+      coordinates: [
+        [
+          [0.80908203125, 0.8935546875],
+          [0.8095703125, 0.89404296875],
+          [0.80908203125, 0.89404296875],
+          [0.80908203125, 0.8935546875]
+        ]
+      ]
+    }
+  }
+];
+
+test('Tile2DHeader#dataInWorldCoordinates', async t => {
+  const [tile, getTileData] = createTile2DHeaderWithMockedData(LOCAL_TILE_COORDS);
+
+  // Mock loader
+  tile._loader = getTileData();
+  // Not loaded data
+  t.is(
+    tile.dataInWorldCoordinates instanceof Promise,
+    tile._loader instanceof Promise,
+    'dataInWorldCoordinates getter correctly returns loader.'
+  );
+  // Load data
+  tile.content = await getTileData();
+  tile._isLoaded = true;
+  // Loaded data
+  t.deepEqual(
+    tile.dataInWorldCoordinates,
+    LOCAL_TILE_COORDS,
+    'dataInWorldCoordinates getter correctly returns no-transformed data.'
+  );
+
+  t.end();
+});
+
+const TRANSFORM_TO_WORLD_VIEWPORT = new WebMercatorViewport({
+  latitude: 0,
+  longitude: 0,
+  zoom: 1
+});
+
+const TRANSFORM_TO_WORLD_BBOX = {west: -180, north: 85.0511287798066, east: 0, south: 0};
+
+test('Tile2DHeader#transformToWorld', async t => {
+  const [tile, getTileData] = createTile2DHeaderWithMockedData(LOCAL_TILE_COORDS);
+
+  tile.bbox = TRANSFORM_TO_WORLD_BBOX;
+  tile.transformToWorld = content =>
+    content.map(object => _transformTileCoordsToWGS84(object, tile, TRANSFORM_TO_WORLD_VIEWPORT));
+  // Mock loader
+  tile._loader = getTileData();
+  // Not loaded data
+  t.is(
```

Comment:
You need to test `await tile.dataInWorldCoordinates`

Reply:
In case data is not loaded `dataInWorldCoordinates` returns a promise, can we check it like this, right?
```javascript
t.is(
    tile.dataInWorldCoordinates instanceof Promise,
    tile._loader instanceof Promise,
    'dataInWorldCoordinates getter correctly returns loader.'
  );
```

Getting data from getter:
```javascript
t.deepEqual(
    await tile.dataInWorldCoordinates,
    tile._transformToWorldDataCache,
    'dataInWorldCoordinates getter correctly returns cached data.'
  );
```

Reply:
You are right, checking